File: math/sum.cpp
#include <stdio.h>

int pow(int n)
{
  if (n >= 1)
    return 2*pow(n-1);
  else
    return 1;
}

int pow_sum(int n)
{
  int sum = 0;
  for (int i = 0; i < n; i++)
    sum += pow(i);
  return sum;
}

int main()
{
  printf("1: %d\n", pow_sum(1));
  printf("2: %d\n", pow_sum(2));
  printf("3: %d\n", pow_sum(3));
  printf("...\n");
  printf("30: %d\n", pow_sum(30));
  printf("31: %d\n", pow_sum(31));
  printf("32: %d\n", pow_sum(32));
}

File: math/sum.py
import numpy as np

def pow(n):
  if (n >= 1):
    return 2*pow(n-1)
  else:
    return np.array(1, dtype=int)

def sum_pow(n):
  sum = np.array(0, dtype=int)
  for i in range(0, n+1, 1):
    sum += pow(i)
  return sum

for i in range(0, 68, 1):
  print(i, sum_pow(i))

